query/src/test/java/org/infinispan/query/startup/IndexStartupModeTest.java

```diff
@@ -0,0 +1,181 @@
+   private final String indexesLocation = CommonsTestingUtil.tmpDirectory("IndexStartupModeTest", "indexes");
+
+   private EmbeddedCacheManager cacheManager;
+   private Cache<String, Developer> cache;
+   private QueryFactory queryFactory;
+
+   public void volatileDataNonVolatileIndexes_purgeAtStartup() {
+      execute(IndexStorage.FILESYSTEM, false, IndexStartupMode.PURGE, () -> {
+         verifyMatches(0, "fax4ever");
+
+         cache.put("fabio", new Developer("fax4ever", "fax@redmail.io", "Infinispan developer", 0));
+
+         verifyMatches(1, "fax4ever");
+      });
+
+      execute(IndexStorage.FILESYSTEM, false, IndexStartupMode.NONE, () -> {
+         // data is not present anymore
+         assertThat(cache.get("fabio")).isNull();
+
+         // but by default no purge is applied on persisted indexes
+         verifyMatches(1, "fax4ever");
+      });
+
+      execute(IndexStorage.FILESYSTEM, false, IndexStartupMode.PURGE, () -> {
+         // with the initial purge the persisted indexes are wiped out at cache startup
+         verifyMatches(0, "fax4ever");
+
+         cache.put("fabio", new Developer("fax4ever", "fax@redmail.io", "Infinispan developer", 0));
+
+         // recreate the index for the next executions
+         verifyMatches(1, "fax4ever");
+      });
+
+      execute(IndexStorage.FILESYSTEM, false, IndexStartupMode.NONE, () -> {
+         // data is not present anymore
+         assertThat(cache.get("fabio")).isNull();
+
+         // but by default no purge is applied on persisted indexes
+         verifyMatches(1, "fax4ever");
+      });
+
+      execute(IndexStorage.FILESYSTEM, false, IndexStartupMode.AUTO, () -> {
+         // auto in this case is equivalent to purge
+         verifyMatches(0, "fax4ever");
+      });
+   }
+
+   public void nonVolatileDataVolatileIndexes_reindexAtStartup() {
+      execute(IndexStorage.LOCAL_HEAP, true, IndexStartupMode.NONE, () -> {
+         verifyMatches(0, "fax4ever");
+
+         cache.put("fabio", new Developer("fax4ever", "fax@redmail.io", "Infinispan developer", 0));
+
+         verifyMatches(1, "fax4ever");
+      });
+
+      execute(IndexStorage.LOCAL_HEAP, true, IndexStartupMode.NONE, () -> {
+         // data is still present
+         Developer fabio = cache.get("fabio");
+         assertThat(fabio).isNotNull();
+         assertThat(fabio).extracting("nick").containsExactly("fax4ever");
+
+         // but indexes have gone!
+         verifyMatches(0, "fax4ever");
+      });
+
+      execute(IndexStorage.LOCAL_HEAP, true, IndexStartupMode.REINDEX, () -> {
+         // data is still present
+         Developer fabio = cache.get("fabio");
+         assertThat(fabio).isNotNull();
+         assertThat(fabio).extracting("nick").containsExactly("fax4ever");
+
+         // reindex on startup is run asynchronously
+         Awaitility.await().untilAsserted(() -> {
```

Comment:
there are `eventually(Condition)` methods available to use. It is not needed to add a new dependency.

Reply:
nice to use the Infinispan utility for that!